go/checks.rst

```diff
@@ -0,0 +1,302 @@
+      },
+      "loopclosure": {
+        "description": "fail builds without exception since we know this check is 100% accurate",
+        "severity": "ERROR"
+      }
+    }
+
+This label referencing this configuration file must be provided as the
+``config`` attribute value of the ``go_checker`` rule.
+
+.. code:: bzl
+
+    go_checker(
+        name = "my_checker",
+        deps = [
+            ":mycheck",
+            ":unsafedom",
+            "@javascript_checks//:loopclosure",
+        ],
+        config = ":config.json"
+        visibility = ["//visibility:public"],
+    )
+
+API
+---
+
+go_checker
+~~~~~~~~~~
+
+This generates a checker program that is run alongside the compiler to analyze
+Go source code.
+
+Providers
+^^^^^^^^^
+
+* GoChecker_
+
+Attributes
+^^^^^^^^^^
+
++----------------------------+-----------------------------+---------------------------------------+
+| **Name**                   | **Type**                    | **Default value**                     |
++----------------------------+-----------------------------+---------------------------------------+
+| :param:`name`              | :type:`string`              | |mandatory|                           |
++----------------------------+-----------------------------+---------------------------------------+
+| A unique name for this rule.                                                                     |
++----------------------------+-----------------------------+---------------------------------------+
+| :param:`deps`              | :type:`label_list`          | :value:`None`                         |
++----------------------------+-----------------------------+---------------------------------------+
+| List of Go libraries that will be linked to the generated checker binary.                        |
+| These libraries must call analysis.Register to ensure that the analyses they implement are       |
+| called by the checker binary.                                                                    |
+| These libraries must be `go_tool_library` rules to avoid bootstrapping problems.                 |
++----------------------------+-----------------------------+---------------------------------------+
+| :param:`config`             | :type:`label`          | :value:`None`                             |
++----------------------------+-----------------------------+---------------------------------------+
+| JSON configuration file that configures one or more of the checks in `deps`.                     |
++----------------------------+-----------------------------+---------------------------------------+
+
+Example
+^^^^^^^
+
+.. code:: bzl
+
+  go_checker(
+        name = "my_checker",
+        deps = [
+            ":mycheck",
+            ":othercheck",
+            "@javascript_checks//:unsafedom", # we can import checks from a remote repo
+        ],
+        config = ":config.json"
+        visibility = ["//visibility:public"],
+    )
+
+
+go_tool_library
```

Comment:
This documentation should probably go in core.rst. It's not strictly part of the checking framework.

Reply:
Done.